sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -867,89 +897,87 @@ export class MessageReceiver extends LinkEntity {
         return;
       }
 
-      // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        const translatedError = translate(receiverError) as MessagingError;
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // Track-1
+      //   - We should only attempt to reopen if either no error was present,
+      //     or the error is considered retryable.
+      // Track-2
+      //  Reopen
+      //   - If no error was present
+      //   - If the error is a MessagingError and is considered retryable
+      //   - Any non MessagingError that was triggered by a disconnect because such errors do not get
+      //     translated by `@azure/core-amqp` to a MessagingError and we need to do our best when
+      //     it comes to connection being disconnected
+      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
+      let shouldReopen = true;
+      if (isMessagingError(translatedError)) {
+        shouldReopen = translatedError.retryable;
+      } else {
+        shouldReopen = !translatedError || causedByDisconnect || false;
```

Comment:
What case are we trying to handle with `false` in the end here?
Non messaging error not caused by disconnect? Why would they be not retryable?

Maybe this should distill down to
```
const shouldReopen = isMessagingError(translatedError) ? translatedError.retryable : true;
```

Reply:
That's a good simplification...

Reply:
Added a `false` since typescript was complaining here about the "undefined" being assigned to the boolean value.

Reply:
> What case are we trying to handle with false in the end here?

Adding `|| false` won't affect anything. No additional case is being handled other than fixing the typescript complaint.

> Non messaging error not caused by disconnect? Why would they be not retryable?

They are not retryable because the `translate()` method did not consider them retryable.

> const shouldReopen = isMessagingError(translatedError) ? translatedError.retryable : true;

Are you suggesting that we should allow all non-messaging errors to be retried?

Reply:
> Adding || false won't affect anything. No additional case is being handled other than fixing the typescript complaint.

Thats not true. There are errors for which translate() will return as is and not change it to a `MessagingError`. For these, you will be setting `shouldReopen` to false, but in Track 1 these will be true

Reply:
Okay, now I see the problem. `shouldReopen` is set to true at the beginning which covered this case. Somehow I went with "They are not retryable because the translate() method did not consider them retryable" and it blinded me all over.

Now I see, why this is good..
> const shouldReopen = isMessagingError(translatedError) ? translatedError.retryable : true;

Reply:
So finally, Going with re-open if
- No error
- Messaging-error that is retryable
- Non-messaging-error

Reply:
Agreed